app/models/miq_websocket_worker.rb

```diff
@@ -0,0 +1,34 @@
+class MiqWebsocketWorker < MiqWorker
+  require_nested :Runner
+
+  REQUIRED_ROLE = 'websocket'
+  self.required_roles = [REQUIRED_ROLE]
+
+  RACK_APPLICATION           = WebsocketServer
+  BALANCE_MEMBER_CONFIG_FILE = '/etc/httpd/conf.d/cfme-balancer-websocket.conf'
+  REDIRECTS_CONFIG_FILE      = '/etc/httpd/conf.d/cfme-redirects-websocket'
+  STARTING_PORT              = 5000
+  PROTOCOL                   = 'ws'
+  LB_METHOD                  = :busy
+  REDIRECTS                  = '/ws'
+  CLUSTER                    = 'evmcluster_websocket'
+
+  def friendly_name
+    @friendly_name ||= "Websocket Worker"
+  end
+
+  def self.validate_config_settings(configuration = VMDB::Config.new("vmdb"))
```

Comment:
And by extension, the constant for the REQUIRED_ROLE is not need (see the changes I made to the Ui worker and WebService worker).